soliket/foreground.py

```diff
@@ -107,10 +181,17 @@ def _get_foreground_model(self,
         # Set component spectra
         self.fg_component_list = {s: self.components[s] for s in requested_cls}
 
-        # Set frequency
-        if not hasattr(freqs, '__len__'):
-            freqs = self.freqs
-            bandint_freqs = self.freqs.copy()
+        # Set exp_ch list
+        if not hasattr(exp_ch, '__len__'):
+            exp_ch = self.exp_ch
+
+        # Set array of freqs to use if bandint_freqs is None
+        if not hasattr(bandint_freqs, '__len__'):
+            if hasattr(eff_freqs, '__len__'):
+                bandint_freqs = np.asarray(eff_freqs)
+            else:
+                self.log.info("no frequency list provided to compute the passbands")
```

Comment:
would this raise an error and stop execution?

Reply:
you're right, I added a `raise LoggedError`